File: C++/1773-count-items-matching-a-rule.cpp
class Solution {
public:
    int countMatches(vector<vector<string>>& items, string ruleKey, string ruleValue) {
      int count = 0;
      int j = ruleKey == "type" ? 0 : ruleKey == "color" ? 1:2;
      for(int i = 0; i < items.size(); i++){
        if(items[i][j] == ruleValue)
          count++;
      }
      return count;
    }
};

File: Java/1773-count-items-matching-a-rule.java
class Solution {
    public int countMatches(List<List<String>> items, String ruleKey, String ruleValue) {
      int count = 0, j;
      if(ruleKey.equals("type"))
        j = 0;
      else if (ruleKey.equals("color"))
        j = 1;
      else
        j = 2;
      for(List<String> list : items){
        if(list.get(j).equals(ruleValue))
          count++;
      }
      return count;
    }
}
